app/javascript/packs/contact-form-navigation-alert.js

```diff
@@ -0,0 +1,23 @@
+$(() => {
+   $("body").attr("data-turbolinks", false);
+
+   const formElement = $("form");
+   const originalState = formElement.serialize();
+   formElement.on("submit", onSubmit)
+
+   function shouldNavigateDirectly() {
+      const currentState = $("form").serialize();
+
+      $("*").blur();
+      return currentState === originalState ? undefined : false;
```

Comment:
Have you tested this on multiple browsers? MDN docs suggest we need to do `e.preventDefault()` and `e.returnValue = ''` https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload#Browser_compatibility but maybe that's outdated

Reply:
Yes, I tested it on IE 11, Chrome, Firefox and Edge